src/ctap/mod.rs

```diff
@@ -118,6 +121,17 @@ pub const ES256_CRED_PARAM: PublicKeyCredentialParameter = PublicKeyCredentialPa
     alg: SignatureAlgorithm::ES256,
 };
 
+// Helpers to perform CBOR read/write while respecting CTAP2 nesting limits.
+fn cbor_read(encoded_cbor: &[u8]) -> Result<cbor::Value, cbor::reader::DecoderError> {
+    cbor::reader::read_nested(encoded_cbor, Some(MAX_CBOR_NESTING_DEPTH))
+}
+fn cbor_write(
+    value: cbor::Value,
+    mut encoded_cbor: &mut Vec<u8>,
+) -> Result<(), cbor::writer::EncoderError> {
+    cbor::writer::write_nested(value, &mut encoded_cbor, Some(MAX_CBOR_NESTING_DEPTH))
```

Comment:
Can we map the error to `Err(Ctap2StatusCode::CTAP2_ERR_VENDOR_INTERNAL_ERROR)`?

Reply:
Done